gcsfs/core.py

```diff
@@ -746,7 +746,9 @@ def created(self, path):
     def _parse_timestamp(self, timestamp):
         assert timestamp.endswith("Z")
         UTC = timezone(timedelta(0))
-        return datetime.fromisoformat(timestamp[:-1]).astimezone(UTC)
+        timestamp = timestamp[:-1]
+        timestamp = timestamp + "0" * (6 - len(timestamp.rsplit(".", 1)[1]))
```

Comment:
```py
timestamp.rjust(desired_length_here, '0')
```
would have worked well here as well

Reply:
That might have been more elegant, but this works too :)